selfdrive/car/mazda/carstate.py

```diff
@@ -0,0 +1,194 @@
+from selfdrive.car.interfaces import CarStateBase
+from selfdrive.car.mazda.values import DBC, LKAS_LIMITS, CAR
+
+class CarState(CarStateBase):
+  def __init__(self, CP):
+    super().__init__(CP)
+
+    can_define = CANDefine(DBC[CP.carFingerprint]['pt'])
+    self.shifter_values = can_define.dv["GEAR"]['GEAR']
+
+    self.cruise_speed = 0
+    self.acc_active_last = False
+    self.low_speed_lockout = True
+    self.low_speed_alert = False
+    self.lkas_on = False
+
+  def update(self, cp, cp_cam):
+
+    ret = car.CarState.new_message()
+    ret.wheelSpeeds.fl = cp.vl["WHEEL_SPEEDS"]['FL'] * CV.KPH_TO_MS
+    ret.wheelSpeeds.fr = cp.vl["WHEEL_SPEEDS"]['FR'] * CV.KPH_TO_MS
+    ret.wheelSpeeds.rl = cp.vl["WHEEL_SPEEDS"]['RL'] * CV.KPH_TO_MS
+    ret.wheelSpeeds.rr = cp.vl["WHEEL_SPEEDS"]['RR'] * CV.KPH_TO_MS
+    ret.vEgoRaw = (ret.wheelSpeeds.fl + ret.wheelSpeeds.fr + ret.wheelSpeeds.rl + ret.wheelSpeeds.rr) / 4.
+    ret.vEgo, ret.aEgo = self.update_speed_kf(ret.vEgoRaw)
+    ret.standstill = ret.vEgoRaw < .1
+
+    speed_kph =  ret.vEgoRaw  // CV.KPH_TO_MS
+
+    can_gear = int(cp.vl["GEAR"]['GEAR'])
+    ret.gearShifter = self.parse_gear_shifter(self.shifter_values.get(can_gear, None))
+
+    ret.leftBlinker = cp.vl["BLINK_INFO"]['LEFT_BLINK'] == 1
+    ret.rightBlinker = cp.vl["BLINK_INFO"]['RIGHT_BLINK'] == 1
+
+    ret.steeringAngle =  cp.vl["STEER"]['STEER_ANGLE']
+    ret.steeringTorque = cp.vl["STEER_TORQUE"]['STEER_TORQUE_SENSOR']
+    ret.steeringPressed = abs(ret.steeringTorque) > LKAS_LIMITS.STEER_THRESHOLD
+
+    ret.steeringTorqueEps = cp.vl["STEER_TORQUE"]['STEER_TORQUE_MOTOR']
+    ret.steeringRate = cp.vl["STEER_RATE"]['STEER_ANGLE_RATE']
+
+    ret.brakePressed = cp.vl["PEDALS"]['BRAKE_ON'] == 1
+    ret.brake = cp.vl["BRAKE"]['BRAKE_PRESSURE']
+    ret.brakeLights = ret.brakePressed
+
+    ret.seatbeltUnlatched = cp.vl["SEATBELT"]['DRIVER_SEATBELT'] == 0
+    ret.doorOpen = any([cp.vl["DOORS"]['FL'], cp.vl["DOORS"]['FR'],
+                        cp.vl["DOORS"]['BL'], cp.vl["DOORS"]['BR']])
+
+    ret.gas = cp.vl["ENGINE_DATA"]['PEDAL_GAS']
+    ret.gasPressed = ret.gas > 1e-3
+
+    # No steer if block signal is on
+    block = cp.vl["STEER_RATE"]['LKAS_BLOCK'] == 1
+
+    # On if no driver torque the last 5 seconds
+    handsoff = cp.vl["STEER_RATE"]['HANDS_OFF_5_SECONDS'] == 1
+
+    # LKAS is enabled at 50kph going up and disabled at 45kph going down
+
+    if speed_kph > LKAS_LIMITS.ENABLE_SPEED:
+      self.lkas_on = True
+    elif speed_kph < LKAS_LIMITS.DISABLE_SPEED and block:
+      self.lkas_on = False
+
+    # if any of the cruize buttons is pressed force state update
+    if any([cp.vl["CRZ_BTNS"]['RES'],
+                cp.vl["CRZ_BTNS"]['SET_P'],
+                cp.vl["CRZ_BTNS"]['SET_M']]):
+      self.acc_active = True
+      self.cruise_speed = ret.vEgoRaw
+
+    ret.cruiseState.available = cp.vl["CRZ_CTRL"]['CRZ_ACTIVE'] == 1
+    if not ret.cruiseState.available:
+      self.acc_active = False
```

Comment:
doesn't need a condition

Reply:
why not ?

Reply:
Only want to sync if cruiseState.available is false. acc_active can be already false but available is true